src/remote.rs

```diff
@@ -613,7 +613,11 @@ impl<'cb> Binding for FetchOptions<'cb> {
                 .map(|m| m.raw())
                 .unwrap_or_else(|| ProxyOptions::new().raw()),
             prune: crate::call::convert(&self.prune),
-            update_flags: self.update_flags.bits() as c_uint,
+            // HACK: `libgit2` uses C bitfields, which do not have a guaranteed memory layout.
+            // Reversing the bits ensures that the bitfields are set whether the bits are laid out
+            // from left to right or right to left, but will not work on other memory layouts.
+            update_flags: (self.update_flags.bits() | self.update_flags.bits().reverse_bits())
```

Comment:
did you check whether this will make the reverse overlap with another meaningful bit?

Reply:
The number of meaningful bits is currently 2, so there will be no overlay (2 at each end, with 4 bits in the middle unused for the 8 bits of an unsigned int). However, this might be an issue if more bits in the flag are used.

Reply:
This is no longer an issue in libgit2 v1.8.1.